shared/src/androidMain/kotlin/io/customer/shared/tracking/queue/QueueTimer.kt

```diff
@@ -0,0 +1,108 @@
+package io.customer.shared.tracking.queue
+
+import android.os.CountDownTimer
+import io.customer.shared.util.Logger
+import io.customer.shared.work.JobDispatcher
+import io.customer.shared.work.JobExecutor
+import io.customer.shared.work.TimeUnit
+import io.customer.shared.work.runOnMain
+
+internal actual fun getQueueTimer(
+    logger: Logger,
+    jobExecutor: JobExecutor,
+): QueueTimer = AndroidQueueTimer(
+    logger = logger,
+    jobExecutor = jobExecutor,
+)
+
+/**
+ * [QueueTimer] implementation that uses Android [CountDownTimer] to run timer countdowns.
+ */
+private class AndroidQueueTimer(
+    private val logger: Logger,
+    override val jobExecutor: JobExecutor,
+) : QueueTimer, JobDispatcher {
```

Comment:
We have encountered bugs in the past with the implementation of our timer in the Android SDK. It makes me nervous to make any modifications to the class unless there are major reasons to do so. 

What changes do you think are worth doing to the Android timer compared to doing a direct copy/paste from the current Android SDK's implementation?

Reply:
I just tried to simplify the same flow and tried to achieve the following

- Remove redundant variables like `timerAlreadyScheduled` as we can simply check this using `null` values
- Cancelling timer directly instead of `Job`, so we need only to store timer reference now

And yeah, thanks for reminding about past bugs on timer that I'm not aware of. Can you share some examples so I can try testing the newer implementation with it? I have tried several tests and it worked fine in almost all cases except that it doesn't get cancelled when app goes in background or closed but stays in memory.

I'm completely okay to revert it to current Android version, but I would really like to know what challenges we faced in the past so if we make any improvements in future, I can take care of them.

Reply:
Makes sense 😁 Thanks for sharing. I'll take note of this so we can take extra care about timer. I'll also see if we can add some tests for this before shipping first version of KMM, just to be sure its working as expected.